src/MassTransit/Logging/Diagnostics/LogContextActivityExtensions.cs

```diff
@@ -17,7 +17,9 @@ public static class LogContextActivityExtensions
             params (string Key, object? Value)[] tags)
             where T : class
         {
-            var activity = Cached.Source.Value.CreateActivity(transportContext.ActivityName, ActivityKind.Producer);
+            var parentActivityContext = GetParentActivityContext(context.Headers, defaultContext: System.Diagnostics.Activity.Current?.Context);
```

Comment:
So, I really don't like the way this is working out.
I really think it needs to be a call without the parentActivityContext if Activity.Current != null. And then let the underlying library do its job. Only even TRY to use the header if the current is null, and the header is present and valid.

Reply:
You're right. Changed to use a default `ActivityContext` (this makes the created activity inherit its parent context) and only set from headers when `Activity.Current` is null.